backend/internal/group/handler/grouphandler.go

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// Package handler provides the implementation for group management operations.
+package handler
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/asgardeo/thunder/internal/group/model"
+	"github.com/asgardeo/thunder/internal/group/provider"
+	"github.com/asgardeo/thunder/internal/system/log"
+)
+
+// GroupHandler is the handler for group management operations.
+type GroupHandler struct {
+}
+
+// HandleGroupListRequest handles the list groups request.
+func (gh *GroupHandler) HandleGroupListRequest(w http.ResponseWriter, r *http.Request) {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, "GroupHandler"))
+
+	groupProvider := provider.NewGroupProvider()
+	groupService := groupProvider.GetGroupService()
+	groups, err := groupService.GetGroupList()
+	if err != nil {
+		http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	err = json.NewEncoder(w).Encode(groups)
+	if err != nil {
+		http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+		return
+	}
+
+	// Log the groups response
+	logger.Debug("Groups GET (list) response sent")
+}
+
+// HandleGroupPostRequest handles the create group request.
+func (gh *GroupHandler) HandleGroupPostRequest(w http.ResponseWriter, r *http.Request) {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, "GroupHandler"))
+
+	var createRequest model.CreateGroupRequest
+	if err := json.NewDecoder(r.Body).Decode(&createRequest); err != nil {
+		http.Error(w, "Bad Request: The request body is malformed or contains invalid data.", http.StatusBadRequest)
+		return
+	}
+
+	// Create the group using the group service
+	groupProvider := provider.NewGroupProvider()
+	groupService := groupProvider.GetGroupService()
+	createdGroup, err := groupService.CreateGroup(createRequest)
+	if err != nil {
```

Comment:
Let's return `ServiceError` from the service layer and handle errors as in https://github.com/asgardeo/thunder/blob/6753309984b727d5fead25e70da98dc232151058/backend/internal/notification/message/handler/notificationhandler.go#L317

Reply:
Fixed with https://github.com/asgardeo/thunder/compare/73946306ccf2e751a0829d5c8c47baf76a871d36..b8ad696888f32b1b4b8f9e6506e6eb438e2b7ed7